File: typescript/functions/awaitTransactionStatus.ts
//トランザクションハッシュを指定してトランザクションの状態を確認する関数
export async function awaitTransactionStatus(
  hash: string,
  nodeUrl: string,
  transactionStatus: "confirmed" | "unconfirmed" | "partial",
) {
  console.log(`${transactionStatus}状態まで待機中..`)
  await new Promise(async (resolve, reject) => {
    for (let i = 0; i < 100; i++) {
      await new Promise((res) => setTimeout(res, 1000))
      //トランザクションハッシュからステータスを確認
      const status = await fetch(
        new URL("/transactionStatus/" + hash, nodeUrl),
        {
            method: 'GET',
            headers: { 'Content-Type': 'application/json' },
        }        
      ).then((res) => res.json())
      //指定したトランザクションステータスになっていたら結果を表示させる
      if (status.group === transactionStatus) {
        console.log("結果 ", status.code, "エクスプローラー ", `https://testnet.symbol.fyi/transactions/${hash}`)
        resolve({})
        return
      } else if (status.group === "failed") {
        console.log("結果　エラー ", status.code)
        resolve({})
        return
      }
    }
    reject(new Error("トランザクションが確認されませんでした。"))
  })
}

File: python/functions/await_transaction_status.py
import time
import requests
from typing import Literal, Dict, Any

# トランザクションハッシュを指定してトランザクションの状態を確認する関数
async def await_transaction_status(
    hash: str,
    node_url: str,
    transaction_status: Literal["confirmed", "unconfirmed", "partial"]
) -> None:
    print(f"{transaction_status}状態まで待機中..")    
    for _ in range(100):
        time.sleep(1)        
        # トランザクションハッシュからステータスを確認
        status: Dict[str, Any] = requests.get(
            f"{node_url}/transactionStatus/{hash}",
            headers={"Content-Type": "application/json"}
        ).json()  
        # 指定したトランザクションステータスになっていたら結果を表示させる        
        if status['code'] == 'ResourceNotFound':
            continue
        elif status['group'] == transaction_status:
            print(f"結果: {status['code']} エクスプローラー: https://testnet.symbol.fyi/transactions/{hash}")
            return
        elif status['group'] == 'failed':
            print("結果 エラー:", status['code'])
            return
            
    raise Exception("トランザクションが確認されませんでした。")